streams/src/test/java/org/apache/kafka/test/InternalProcessorContextMock.java

```diff
@@ -0,0 +1,376 @@
+            final Capture<Object> keyCapture = Capture.newInstance();
+            final Capture<Object> valueCapture = Capture.newInstance();
+            final Capture<To> toCapture = Capture.newInstance();
+
+            mock.forward(capture(keyCapture), capture(valueCapture), capture(toCapture));
+            expectLastCall().andAnswer(() -> {
+                final To to = toCapture.getValue();
+                processorContext.forward(keyCapture.getValue(), valueCapture.getValue(), to);
+                toInternal.update(to);
+                if (toInternal.hasTimestamp()) {
+                    setRecordContext(new ProcessorRecordContext(toInternal.timestamp(), mock.offset(), mock.partition(), mock.topic(), mock.headers()));
+                }
+                final ProcessorNode thisNode = processorNode;
+                try {
+                    for (final ProcessorNode childNode : (List<ProcessorNode<K, V>>) thisNode.children()) {
+                        if (toInternal.child() == null || toInternal.child().equals(childNode.name())) {
+                            processorNode = childNode;
+                            childNode.process(keyCapture.getValue(), valueCapture.getValue());
+                            toInternal.update(to); // need to reset because MockProcessorContext is shared over multiple Processors and toInternal might have been modified
+                        }
+                    }
+                } finally {
+                    processorNode = thisNode;
+                }
+                return null;
+            }).anyTimes();
+        }
+
+        private void schedule() {
+            final Capture<Duration> interval = Capture.newInstance();
+            final Capture<PunctuationType> type = Capture.newInstance();
+            final Capture<Punctuator> punctuator = Capture.newInstance();
+            expect(mock.schedule(capture(interval), capture(type), capture(punctuator)))
+                    .andAnswer(() -> processorContext.schedule(interval.getValue(), type.getValue(), punctuator.getValue()))
+                    .anyTimes();
+        }
+
+        private void getStateStore() {
+            final Capture<String> stateStoreNameCapture = Capture.newInstance();
+            expect(mock.getStateStore(capture(stateStoreNameCapture)))
+                    .andAnswer(() -> stateStoreMap.get(stateStoreNameCapture.getValue()))
+                    .anyTimes();
+        }
+
+        private void register() {
+            final Capture<StateStore> storeCapture = Capture.newInstance();
+            final Capture<StateRestoreCallback> restoreCallbackCapture = Capture.newInstance();
+
+            mock.register(capture(storeCapture), capture(restoreCallbackCapture));
+
+            expectLastCall()
+                    .andAnswer(() -> {
+                        stateStoreMap.put(storeCapture.getValue().name(), storeCapture.getValue());
+                        return null;
```

Comment:
Q: Do you think we need to implement this setter behaviour for this mock? Couldn't we not just pass a map of state stores at during the build specification with a method like `stateStores(final Map<String, StateStore> stateStores)` and don't specify a behaviour for `register()`?

Reply:
Hi @cadonna,
I didn't get the advantage of this. 
I would point out that `register()` is often called here: [InMemoryWindowStore.java#L111](https://github.com/apache/kafka/blob/2.4/streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java#L111)

Anyway, I made all requested modifications except this one, could you take another look?
Thank you.

Reply:
@pierDipi Sorry for the late reply.
Actually, during our discussions, I thought we agreed to stop the EasyMock experiment because of the public API issue and we should try to do something like #7979 as proposed by @vvcephei. Sorry for the confusion. 
Anyways, I would be really happy, if you could take #7979 and extend it to make it robust. Are you still interested?

Reply:
@cadonna Yes, I'm still interested, but I cannot guarantee when it will happen. I hope it will be done by the end of the next month.